jiant/tasks/qa.py

```diff
@@ -441,3 +446,165 @@ def get_metrics(self, reset=False):
             self._score_tracker = collections.defaultdict(list)
 
         return {"f1": f1, "em": em, "avg": (f1 + em) / 2}
+
+
+@register_task("qasrl", rel_path="QASRL/")
+class QASRLTask(SpanPredictionTask):
+    def __init__(self, path, max_seq_len, name, **kw):
+        """QA-SRL (Question-Answer Driven Semantic Role Labeling)
+        See http://qasrl.org/
+        Download, unzip, and rename the "qasrl-v2" folder to "QASRL"
+        """
+        super(QASRLTask, self).__init__(name, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+
+        self.train_data = None
+        self.val_data = None
+        self.test_data = None
+
+        self.f1_scorer = SpanF1Measure()
+        self.val_metric = "%s_f1" % self.name
+        self.val_metric_decreases = False
+
+    def get_metrics(self, reset=False):
+        """Get metrics specific to the task"""
+        f1 = self.f1_scorer.get_metric(reset)
+        return {"f1": f1}
+
+    def update_metrics(self, logits, labels, tagmask=None):
+        self.f1_scorer(
+            pred_start=logits["span_start"],
+            pred_end=logits["span_end"],
+            gold_start=labels["span_start"],
+            gold_end=labels["span_end"],
+        )
+
+    def load_data(self):
+        self.train_data = self._load_file(os.path.join(self.path, "orig", "train.jsonl.gz"))
+
+        self.val_data = self._load_file(os.path.join(self.path, "orig", "dev.jsonl.gz"))
+        self._shuffle_data(self.val_data)
+
+        self.test_data = self._load_file(os.path.join(self.path, "orig", "test.jsonl.gz"))
+        self.sentences = []
+
+        self.sentences = (
+            self.train_data[0] + self.train_data[1] + self.val_data[0] + self.val_data[1]
+        )
+
+    def get_sentences(self) -> Iterable[Sequence[str]]:
+        """ Yield sentences, used to compute vocabulary. """
+        yield from self.sentences
+
+    def process_split(
+        self, split, indexers, model_preprocessing_interface
```

Comment:
what is `model_preprocessing_interface`?

Reply:
It seems to be a new addition following the new integration of pytorch_transformers, for handling the different model types.